zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -546,12 +549,26 @@ where
                     .memo
                     .as_ref()
                     .map_or_else(MemoBytes::empty, |m| m.clone());
-                builder.add_sapling_output(
-                    external_ovk,
-                    *ua.sapling().expect("TODO: Add Orchard support to builder"),
-                    payment.amount,
-                    memo.clone(),
-                )?;
+
+                if let Some(sapling_receiver) = ua.sapling() {
+                    builder.add_sapling_output(
+                        external_ovk,
+                        *sapling_receiver,
+                        payment.amount,
+                        memo.clone(),
+                    )?;
+                } else if let Some(taddr) = ua.transparent() {
+                    if payment.memo.is_some() {
+                        return Err(Error::MemoForbidden);
+                    } else {
+                        builder.add_transparent_output(
+                            taddr, 
+                            payment.amount
+                        )?;
+                    }
+                } else {
+                    return Err(Error::UnsupportedPoolType(PoolType::Shielded(ShieldedProtocol::Orchard)));
+                }
                 sapling_output_meta.push((
                     Recipient::Unified(ua.clone(), PoolType::Shielded(ShieldedProtocol::Sapling)),
```

Comment:
This is incorrect if we took the `builder.add_transparent_output(...)` branch above. Blocking.